app/uk/gov/hmrc/breathingspaceifstub/controller/UnderpaymentsController.scala

```diff
@@ -82,7 +82,13 @@ class UnderpaymentsController @Inject()(underpaymentsService: UnderpaymentsServi
         .get(nino, periodId)
         .map(
           _.fold(
-            _ => logAndGenFailureResult(Failure(NO_DATA_FOUND)),
+            failure => {
+              if (failure.baseError != BaseError.GATEWAY_TIMEOUT) {
```

Comment:
This is a bit clunky. Make it more readable with a pattern match expression.

Reply:
Fixed